### PR TITLE
lesson tasks and hometasks for lesson 37-38

### PR DESCRIPTION
**Pull request by marivdau**

### CODE REVIEW

File: fe49-onl-maryia-dauhan/src/App.tsx [hunk 1]
```diff
@@ -1,24 +1,67 @@
...
-        >
-          Learn React
-        </a>
-      </header>
+    <div>
+      <Header />
+
+      <Title>Lesson task #37 and Hometask #37</Title>
+
+      <Title>Sign In</Title>
```
Comment (svr93) [inline fe49-onl-maryia-dauhan/src/App.tsx:19]: Multiple h1 tags on the page is the semantic problem :(

File: fe49-onl-maryia-dauhan/src/features/tab/tabs.tsx [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+  };
+
+  return (
+    <TabWrapper>
+      <TabList>
+        {tabs.map((tab, id) => (
+          <TabItem>
+            <TabButton
+              type='button'
+              key={id}
```
Comment (svr93) [inline fe49-onl-maryia-dauhan/src/features/tab/tabs.tsx]: key should be placed on the first root element inside callback (`<TabItem />`)

File: fe49-onl-maryia-dauhan/src/features/hamburger/hamburger.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+          className='burger1'
+          style={{ transform: hamburgerOpen ? 'rotate(45deg)' : 'rotate(0)' }}
+        ></Burger>
+        <Burger
+          className='burger2'
+          style={{ transform: hamburgerOpen ? 'rotate(-45deg)' : 'rotate(0)' }}
+        ></Burger>
+      </Humburger>
+
+      <ListWrapper style={{ display: hamburgerOpen ? 'block' : 'none' }}>
```
Comment (svr93) [inline fe49-onl-maryia-dauhan/src/features/hamburger/hamburger.tsx:32]: the logic should be placed inside styled component

File: fe49-onl-maryia-dauhan/src/ui/postcard/short-postcard/short-postcard.tsx [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+      <ShortSecondLine>
+        <ShortLikeDiv>
+          <VoteButton type='button' onClick={() => setVoteUp(voteUp + 1)}>
+            <ShortActionImage
+              alt='like'
+              src={require('../../../images/like-svgrepo-com.svg').default}
+            />
+          </VoteButton>
+          <ShortActionCounter>{voteUp}</ShortActionCounter>
+          <VoteButton type='button' onClick={() => setVoteDown(voteDown + 1)}>
```
Comment (svr93) [inline fe49-onl-maryia-dauhan/src/ui/postcard/short-postcard/short-postcard.tsx:55]: user state should be checked too: if you already voted up you should `-1` from voteUp state & add `+1` to voteDown

File: fe49-onl-maryia-dauhan/src/ui/postcard/medium-postcard/medium-postcards.tsx [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+  title,
+  author,
+}) => {
+  const formatDate = (dateString: string) => {
+    const options: Intl.DateTimeFormatOptions = {
+      year: 'numeric',
+      month: 'long',
+      day: 'numeric',
+    };
+    return new Date(dateString).toLocaleDateString([], options);
```
Comment (svr93) [inline fe49-onl-maryia-dauhan/src/ui/postcard/medium-postcard/medium-postcards.tsx:29]: wonderful

**Review by svr93 — APPROVED** [review]

File: fe49-onl-maryia-dauhan/src/App.tsx (hunk 1 above)
Comment (marivdau) [reply fe49-onl-maryia-dauhan/src/App.tsx:19]: fixed

File: fe49-onl-maryia-dauhan/src/features/tab/tabs.tsx (hunk 2 above)
Comment (marivdau) [reply fe49-onl-maryia-dauhan/src/features/tab/tabs.tsx]: fixed. thank you

File: fe49-onl-maryia-dauhan/src/features/hamburger/hamburger.tsx (hunk 3 above)
Comment (marivdau) [reply fe49-onl-maryia-dauhan/src/features/hamburger/hamburger.tsx:32]: fixed

File: fe49-onl-maryia-dauhan/src/ui/postcard/short-postcard/short-postcard.tsx (hunk 4 above)
Comment (marivdau) [reply fe49-onl-maryia-dauhan/src/ui/postcard/short-postcard/short-postcard.tsx:55]: added logic to default postcard
image

File: fe49-onl-maryia-dauhan/src/ui/postcard/medium-postcard/medium-postcards.tsx (hunk 5 above)
Comment (marivdau) [reply fe49-onl-maryia-dauhan/src/ui/postcard/medium-postcard/medium-postcards.tsx:29]: thank you

File: fe49-onl-maryia-dauhan/src/ui/postcard/short-postcard/short-postcard.tsx (hunk 4 above)
Comment (marivdau) [reply fe49-onl-maryia-dauhan/src/ui/postcard/short-postcard/short-postcard.tsx:55]: same for voting down

Comment (marivdau) [reply fe49-onl-maryia-dauhan/src/ui/postcard/short-postcard/short-postcard.tsx:55]: @svr93

**Pull request merged**